libraries/AP_Scripting/examples/plane-doublets.lua

```diff
@@ -48,109 +46,78 @@ function retry_set_mode(mode)
-        elseif now < start_time + (DOUBLET_TIME * 2) then
+            gcs:send_text(6, "pwm" .. tostring(SRV_Channels:get_output_pwm(DOUBLET_FUCNTION)))
+        elseif now < (start_time + DOUBLET_TIME) and now > (start_time + (DOUBLET_TIME / 2))then
+            gcs:send_text(6, "DOUBLET down" .. doublet_srv_chan)
+            down = doublet_srv_trim - math.floor((doublet_srv_trim - doublet_srv_min) * (DOUBLET_MAGNITUDE / 45))
+            SRV_Channels:set_output_pwm_chan_timeout(doublet_srv_chan, down, DOUBLET_TIME / 2 + 100)
+            gcs:send_text(6, "pwm" .. tostring(SRV_Channels:get_output_pwm(DOUBLET_FUCNTION)))
+        elseif now > start_time + (DOUBLET_TIME * 2) then
             -- stick fixed response
             -- hold at pre doublet trim position for any damping effects
-            SRV_Channels:set_output_pwm_chan_timeout(doublet_srv_chan, doublet_srv_trim, DOUBLET_TIME * 2)
-        elseif now > start_time + (DOUBLET_TIME * 2) then
-            -- notify GCS
-            end_time = now
-            gcs:send_text(6, "DOUBLET FINISHED")
-        else
-            gcs:send_text(6, "this should not be reached")
-        end
-    elseif end_time ~= -1 and rc:get_pwm(DOUBLET_ACTION_CHANNEL) > 1700 then
-        -- wait for RC input channel to go low
-        gcs:send_text(6, "RC" .. DOUBLET_ACTION_CHANNEL .. " still high")
-        callback_time = 100 -- prevent spamming messages to the GCS
-    elseif now ~= -1 and end_time ~= -1 then
-        gcs:send_text(6, "RETURN TO PREVIOUS FLIGHT MODE")
-        now = -1
-        end_time = -1
-        start_time = -1
-        -- clear all of the timeouts
-        control_functions = {K_AILERON, K_ELEVATOR, K_THROTTLE, K_RUDDER}
-        for i = 1, 4 do
-            local control_chan = SRV_Channels:find_channel(control_functions[i])
-            SRV_Channels:set_output_pwm_chan_timeout(control_chan, param:get("SERVO" .. control_chan + 1 .. "_TRIM"), 0)
-        end
-        retry_set_mode(pre_doublet_mode)
-        callback_time = 100 -- don't need to rerun for a little while
-    elseif now ~= -1 then
-        -- stopped before finishing. recover to level attitude
-        gcs:send_text(6, "FBWA RECOVER")
-        now = -1
-        end_time = -1
-        start_time = -1
-        -- clear all of the timeouts
-        control_functions = {K_AILERON, K_ELEVATOR, K_THROTTLE, K_RUDDER}
-        for i = 1, 4 do
-            local control_chan = SRV_Channels:find_channel(control_functions[i])
-            SRV_Channels:set_output_pwm_chan_timeout(control_chan, param:get("SERVO" .. control_chan + 1 .. "_TRIM"), 0)
+            gcs:send_text(6, "DOUBLET finished")
+
+            if vehicle:get_mode() == MODE_MANUAL then
+                retry_set_mode(MODE_FBWA)
+                gcs:send_text(6, "DOUBLET FINISHED, in mode" ..vehicle:get_mode())
+                SRV_Channels:set_output_pwm_chan_timeout(DOUBLET_ACTION_CHANNEL, 1000, DOUBLET_TIME)
```

Comment:
Your outputting  a servo using the RC input channel?

Reply:
If you swap to `rc:get_aux_cached(300)` for the input you can reset it by calling ` rc:run_aux_function(300, 0)`